src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeEventService.kt

```diff
@@ -37,4 +42,20 @@ class BarcodeEventService(private val barcodeEventRepository: BarcodeEventReposi
         )
         throw ValidationException(Duplicate(firstCheck.createdDateTime, firstCheck.location))
       }
+
+  fun checkForExpired(barcode: Barcode, userId: String, location: String) =
+    barcodeEventRepository.findByBarcodeAndStatusOrderByCreatedDateTime(barcode, BarcodeStatus.CREATED)
+      .firstOrNull()
+      ?.takeIf { createdEvent -> createdEvent.createdDateTime < Instant.now().minus(barcodeConfig.expiry) }
```

Comment:
Is there a (remote) edge case here that the barcode expiry could be xDays, then a barcode is created on day 1 (and hence its expiry is 1+x), but before it is scanned and checked (ie. its in the physical postal system) the barcode expiry config is updated to yDays. Therefore when the barcode is scanned we dont know its real expiry date - all we know is its creation date and the current expiry config.

I know its a real edge case, but I don't know if its an issue? If we think its a issue I guess it means a new column on the Barcode table of its expiry date, that is computed at the point of creation ?

Reply:
It's a really good point... though it is very edge case. And also very hard to predict what the requirement would be if the expiry duration changed - they might be happy to apply the new rule retrospectively anyway. BUT if we go with your suggestion then at least we can be flexible either way.

I'm going to have to sleep on this one...